src/Draco.Compiler/Api/Diagnostics/Diagnostic.cs

```diff
@@ -37,9 +37,29 @@ public sealed record class DiagnosticTemplate
     /// <param name="title">A short title for the message.</param>
     /// <param name="severity">The severity of the message.</param>
     /// <param name="format">The format string that describes the diagnostic in detail.</param>
+    /// <param name="errorCode">The error code of the error.</param>
     /// <returns>The constructed <see cref="DiagnosticTemplate"/>.</returns>
-    public static DiagnosticTemplate Create(string title, DiagnosticSeverity severity, string format) =>
-        new(title: title, severity: severity, format: format);
+    public static DiagnosticTemplate Create(string title, DiagnosticSeverity severity, string format, string errorCode) =>
```

Comment:
Let's make error code the first, since that's the identifier of error types from now on